### PR TITLE
Model versioning to accommodate 2025b release

### PR DESCRIPTION
**Pull request by n-clinton**
Add model versioning as new directory structure under models/.  Add the 2025b content including images and technical documentation.

### CODE REVIEW

File: models/model_2025b/README.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
+# Introduction
+
+Here, we introduce the first pan-tropical, high resolution (10-meter) dataset that maps four major commodity tree crops: cocoa, coffee, rubber, and oil palm. For this work, we assembled a large, community-sourced reference database of over eight million points, integrating field observations and expert interpretation of high-resolution (less than one meter) imagery. We combined these samples with the Google Satellite Embeddings dataset in a deep learning framework that generates both classification labels and per-pixel class probability layers. Unlike existing, static map products, our model is deployable on-demand in Google Earth Engine (Gorelick et al., 2017), allowing us to generate maps for any year and region of interest. As continuous per-pixel probability layers, the dataset offers global consistency, local customization and per-pixel entropy as a spatial measure of model confusion. We illustrate the utility of our maps by showcasing representative outputs in key production regions and demonstrating how the dataset can be adapted to improve local performance and track change dynamics.
```
Comment (brycecr) [inline models/model_2025b/README.md]: nit (we can do this later): adding a link for Satellite Embeddings would be nifty.

Comment (brycecr) [inline models/model_2025b/README.md]: "per-pixel entropy" — this seems like a stretch for the README to me? Reading this as a new user I would presume there was some sort of "entropy" band

File: models/model_2025b/README.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+# Methods
+
+## Geospatial foundation model inputs
+
+As predictive inputs to the model, we used the AlphaEarth geospatial foundation model embeddings (Brown et al. 2025).  These embeddings constitute a 64-dimensional feature space, at 10 meters spatial resolution, produced annually.  The feature space contains condensed information from a variety of sensors, spectral regions, and derivative products such as climate and topography and was designed to facilitate predictive modeling of land surface characteristics in "data scarce regimes."  We use these data as predictive features on a per-pixel basis.  Because information from a 1280x1280 meter patch and a year's worth of data is encoded in the 64-D vectors, spatiotemporal convolutions are unnecessary in the model.  We used these data directly as predictive features and also as a change detector, during temporal augmentation.
```
Comment (brycecr) [inline models/model_2025b/README.md]: nit (can be fixed later): "AlphaEarth geospatial foundation model embeddings" IIUC this isn't quite aligned with the preferred terminology. AlphaEarth Foundations Satellite Embeddings might be better

File: models/model_2025b/README.md [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+
+In aggregate, the commodity database contains 12,676,728 points, representing a wide variety of production systems across pantropical regions, including both smallholder and industrial operations, varying levels of shade tree density, monocultures and mixed agroforestry systems, and diverse management types, such as terraced, pivot-irrigated, and mixed-age stands (Figure 2).  The reference data also span multiple years: 2017-2024, and are temporally imbalanced by class, with some years and classes in relatively high proportion in the database.
```
Comment (brycecr) [inline models/model_2025b/README.md]: nit: slightly odd punctuation here

Comment (brycecr) [reply models/model_2025b/README.md]: for 2025b an effort was made to balance the data, no?

File: models/model_2025b/README.md [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+## Reference polygon sampling
+
+Some reference data was non-specific, with geometries containing more than the labeled commodity (e.g. roads, buildings, windbreaks, patches of other vegetation).  To improve sample purity, we masked commodity data with the ancillary datasets to exclude areas of natural forest, bare land, or land cover without the presence of the labeled commodity.  To capture within-field variability arising from different canopy densities, planting ages, and irregular spatial arrangements, we buffered polygons by 10 meters and systematically sampled unmasked pixels with a lattice at 10 or 20 meters spacing.
```
Comment (brycecr) [inline models/model_2025b/README.md]: nit: We could clarify that we applied these buffering & masking steps on an as-needed basis

File: models/model_2025b/README.md [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+
+## Negatives sampling
+
+To create "forest" pseudo-absences, within one degree of each commodity presence point, we randomly selected one "forest" sample from pixels classified as both natural and persistent forest (natural forest probability greater than 0.9 and forest persistence probability greater than 0.95).  To create "other" pseudo-absences, within one degree of each commodity presence point, we randomly selected one "other" sample from pixels classified as not trees, shrubs or mangroves in both Dynamic World and Worldcover.  For both "other" and "forest" pseudo-absences, we also excluded pixels previously mapped as commodities with high confidence.  The date of ancillary datasets (2020, except Dynamic World) resulted in a pseudo-absences dataset corresponding to 2020.
```
Comment (brycecr) [inline models/model_2025b/README.md]: is the forest persistence value description as "probability" intended? Would it be better to refer to it as a score?

Comment (brycecr) [reply models/model_2025b/README.md]: I think we can clarify here how many pseudo-absences were selected (we balanced absences & presence, IIRC?)

File: models/model_2025b/README.md [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+| Cocoa | 1,188,945 |
+| Coffee | 3,429,946 |
+| Palm | 1,282,745 |
+| Rubber | 1,452,286 |
+| Natural forest | 2,486,569 |
+| Other | 2,836,237 |
+
+## Spatial partitioning
+
+For training, model selection and accuracy assessment, we required three independent sets: training, test and evaluation respectively. We assume that labeled points are independent if they are more than 640 meters apart, or half the 1280 meter receptive field of the Alpha Earth foundations model we used for prediction. Although we tolerate spatial autocorrelation in the training data (i.e. points closer than 640 meters), in the test and evaluation sets, points are more than 640 meters apart.
```
Comment (brycecr) [inline models/model_2025b/README.md]: nit: extra 'respectively'. I also think we could clarify the bit about spatial autocorrelation in training (e.g. in order to capture signal for within-field variation, points in the training set may be close together, but in the test + eval sets all points are more than 640 meters apart from all other points, including from training data)

File: models/model_2025b/README.md [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
..., we systematically sampled the reference database (pooled with pseudo-absences) globally. Systematic sampling prioritizes geographic representativeness, ensuring that all major production regions, cropping systems, and ecological zones are represented in the sample. In the square degrees containing any training data, we created a lattice of 0.025 degree (~2780 meters) resolution (n=1,846,400). We sampled the reference database with this lattice by selecting the closest point in the pooled reference data and pseudo-absences database that is within 1280 meters of each lattice point, if such a point existed. We partitioned this hold-out sample (n=215,140) into two assumed independent sets (test and evaluation) using a checkerboard pattern to minimize inter and infra sample spatial autocorrelation. To minimize spatial autocorrelation between the training data and any set used for accuracy assessment, we pooled the hold-out set locations with locations from Descals (2024a,  2024b), which we use as independent validation sets for palm and rubber, and removed all data from the training set within 640 meters of any point in the test set, evaluation set or the Descals (2024a,  2024b) sets.
```
Comment (brycecr) [inline models/model_2025b/README.md:113]: nit: is "n" only over land in the pantropical region?

Comment (brycecr) [reply models/model_2025b/README.md:113]: Do we also remove data close to shiel?

File: models/model_2025b/README.md [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+
+Through this process, we created three assumed independent sets: a training set, a test (model selection) set, and an evaluation set. We used Descals (2024a, n=18,812) for independent validation of the palm model and Descals (2024b, n=1555) for independent validation of the rubber model. Neither the evaluation set nor either of the Descals validation sets were used until after model selection, during the preparation of this manuscript. Table 3 shows the class distribution of training, test and evaluation data.
```
Comment (brycecr) [inline models/model_2025b/README.md:115]: What about Shiel?

File: models/model_2025b/README.md [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+| Rubber | 783,491 |
+| Natural forest | 807,848 |
+| Other | 794,881 |
+
+**Figure 3. Label count by year and class after temporal augmentation..**
+![balanced_training_data_distribution_2025b](/assets/images/balanced_training_data_distribution_2025b.png)
+
+## Classification
+
+Model training and classification were performed using a cloud-based infrastructure similar to Google Cloud AutoML (Bisong et al., 2019). Model architecture, activation functions, optimization routines, learning rates, batch sizes, and training iterations were selected through an automated hyperparameter search, with up to 24 hours allocated for optimization and 24 hours for training. Model architecture is constrained to deep neural networks (DNNs), or networks with one or more hidden layers and one-dimensional inputs. We used the test set for model selection in the optimization phase, the training set for training and did not use the evaluation set.
```
Comment (brycecr) [inline models/model_2025b/README.md]: nit: probably not necessary to add the bit about hidden layers and one-dimmensional inputs. IIUC it's not a precise definition of DNNs and as such might just confuse the issue.

File: models/model_2025b/README.md [hunk 10]
```diff
@@ -0,0 +1,267 @@
...
+The outputs generated by this study are: 
+
+1. Fully trained tree crop classification models, deployable on-the-fly in Google Earth Engine via VertexAI.
+2. Per-pixel probability layers for each class for selected even-numbered years.
+
+These products are designed to flexibly adapt to diverse use cases, for example by tuning per-class thresholds to local conditions and/or independent field data. The probability layers, when stacked with other datasets, form a per-pixel classification. For example, when combined with natural forest (Neumann et al.) and water from WorldCover (Zanaga et al. 2021), a pan-tropical map of commodity production in 2020 is assembled.  
+
+Data Link: [https://goo.gle/fodapa-layers-2025b](https://goo.gle/fodapa-layers-2025b)
+
+## Technical validation
```
Comment (brycecr) [inline models/model_2025b/README.md:194]: nit: empty for now?

File: models/model_2025b/README.md [hunk 11]
```diff
@@ -0,0 +1,267 @@
...
+| Coffee | 695,247 |
+| Palm | 822,462 |
+| Rubber | 783,491 |
+| Natural forest | 807,848 |
+| Other | 794,881 |
+
+**Figure 3. Label count by year and class after temporal augmentation..**
+![balanced_training_data_distribution_2025b](/assets/images/balanced_training_data_distribution_2025b.png)
+
+## Classification
```
Comment (brycecr) [inline models/model_2025b/README.md:177]: nit: We're a bit unclear here about multiclass vs single class models. For the purposes of this readme, multiclass models aren't relavent except as a step on the way to single class models.

File: models/model_2025b/README.md [hunk 12]
```diff
@@ -0,0 +1,267 @@
...
+The final TensorFlow-based DNN was hosted on Vertex AI and deployed in Google Earth Engine. The model outputs probability of class occurrence, either in a vector, for multiclass training, or as a scalar for single class model training.
+
+## Data Records 
+
+The outputs generated by this study are: 
+
+1. Fully trained tree crop classification models, deployable on-the-fly in Google Earth Engine via VertexAI.
+2. Per-pixel probability layers for each class for selected even-numbered years.
+
+These products are designed to flexibly adapt to diverse use cases, for example by tuning per-class thresholds to local conditions and/or independent field data. The probability layers, when stacked with other datasets, form a per-pixel classification. For example, when combined with natural forest (Neumann et al.) and water from WorldCover (Zanaga et al. 2021), a pan-tropical map of commodity production in 2020 is assembled.
```
Comment (brycecr) [inline models/model_2025b/README.md]: nit: ensembled instead of stacked? to align with our terminology in the demo app. Also not sure I quite follow joining with forests + water as commodity production map. In theory we could just join all the commodity layers, no?

File: models/model_2025b/README.md [hunk 13]
```diff
@@ -0,0 +1,267 @@
...
+
+*   **Coffee (generated evaluation):** 0.7 @ 0.28
+![coffee_test_accuracy_2025b](/assets/images/coffee_test_accuracy_2025b.png)
+
+## Limitations
+
+*   Comparison of estimates of accuracy from generated evaluation data and independent validation data suggests that accuracies from generated evaluation data are overestimates. It is incumbent upon the consumer of the data to conduct localized accuracy assessments to ensure suitability for any particular purpose.
+*   Qualitative inspection of the categorical maps revealed mixed performance across regions. 
+    *   Palm is overestimated in dense canopy forest.
+    *   Coffee is overestimated in topographically suitable locations.
```
Comment (brycecr) [inline models/model_2025b/README.md]: nit: would it be useful to be more specific here, using the list sent to TT?

File: models/model_2025b/README.md [hunk 14]
```diff
@@ -0,0 +1,267 @@
...
+    *   Rubber is underestimated in South Sumatra.
+*   While the tree crops were all overpredicted on average, many areas showed strong correspondence with existing plantation locations and field boundaries. 
+*   Our training dataset is largely opportunistic, and does not comprehensively represent every class in every region. Users should exercise extreme caution when using data in areas not represented in our training dataset.
+*   Temporal instability between years is present. Determination of change may require statistical comparison of multiple time periods (Clinton et al. 2024), adjustment of thresholds annually, or post processing such as BULC (Cardille and Fortin 2016).
+*   Input artifacts may be visually apparent in output probabilities and result in classification errors at some thresholds.
```
Comment (brycecr) [inline models/model_2025b/README.md:229]: nit: this is probably true but less of an issue than in the past (maybe worth calling out if we keep this point?)

File: models/model_2025b/README.md [hunk 15]
```diff
@@ -0,0 +1,267 @@
...
+-   Vollrath, Andreas, Jennifer Adams, Sara Aparicio, and John Mrziglod. A global palm oil map for the year 2017 using multi-sensor sar imagery. 2019. Living Planet Symposium. ESA/ESRIN, Frascati, Italy.
+-   Wang, Y., Hollingsworth, P.M., Zhai, D. et al. High-resolution maps show that rubber causes substantial deforestation. 2023. Nature 623, 340–346. <https://doi.org/10.1038/s41586-023-06642-z>
+-   Zanaga, D., Van De Kerchove, R., De Keersmaecker, W., Souverijns, N., Brockmann, C., Quast, R., Wevers, J., Grosu, A., Paccini, A., Vergnaud, S., Cartus, O., Santoro, M., Fritz, S., Georgieva, I., Lesiv, M., Carter, S., Herold, M., Li, Linlin, Tsendbazar, N.E., Ramoino, F., Arino, O., 2021. ESA WorldCover 10 m 2020 v100. <https://doi:10.5281/zenodo.5571936>
+
+# Suggested citation
+
+Forest Data Partnership. 2026. Community models 2025b. https://github.com/google/forest-data-partnership/edit/main/models/model_2025a/README.md
```
Comment (brycecr) [inline models/model_2025b/README.md]: fix url: 2025b

**Review by brycecr — CHANGES_REQUESTED** [review]
I left a number of comments, most of which are minor copy / text changes. There are a few that we should fix before merging IMO:

* url for citation
* clarifying Descals/Shiel rubber

I can fix the first of these and will follow up with you on the second

File: models/model_2025b/README.md (hunk 2 above)
Comment (aSullivan-geo) [reply models/model_2025b/README.md]: That's correct, the naming is "AlphaEarth Foundations Satellite Embedding dataset"

File: models/model_2025b/README.md (hunk 1 above)
Comment (brycecr) [reply models/model_2025b/README.md]: done

File: models/model_2025b/README.md (hunk 2 above)
Comment (brycecr) [reply models/model_2025b/README.md]: done

File: models/model_2025b/README.md (hunk 4 above)
Comment (brycecr) [reply models/model_2025b/README.md]: ah, this was already pretty clear on a second read, but I tweaked it slightly to clarify further

File: models/model_2025b/README.md (hunk 5 above)
Comment (brycecr) [reply models/model_2025b/README.md]: addressed both of these

File: models/model_2025b/README.md (hunk 6 above)
Comment (brycecr) [reply models/model_2025b/README.md]: done

File: models/model_2025b/README.md (hunk 1 above)
Comment (n-clinton) [reply models/model_2025b/README.md]: Done.

Comment (n-clinton) [reply models/model_2025b/README.md]: It's now "options to measure model confusion."

File: models/model_2025b/README.md (hunk 2 above)
Comment (n-clinton) [reply models/model_2025b/README.md]: Done.

File: models/model_2025b/README.md (hunk 3 above)
Comment (n-clinton) [reply models/model_2025b/README.md]: Yes, this is just describing the imbalanced reference dataset.  Feel free to suggest punctuation improvements.

File: models/model_2025b/README.md (hunk 4 above)
Comment (n-clinton) [reply models/model_2025b/README.md]: Done.  "The buffering and/or masking were applied as-needed to select datasets obviously impure on visual inspection."

File: models/model_2025b/README.md (hunk 11 above)
Comment (n-clinton) [reply models/model_2025b/README.md:177]: Done.  The multiclass stuff is gone.

File: models/model_2025b/README.md (hunk 12 above)
Comment (n-clinton) [reply models/model_2025b/README.md]: Done.

File: models/model_2025b/README.md (hunk 13 above)
Comment (n-clinton) [reply models/model_2025b/README.md]: I made it more like the TT list.

File: models/model_2025b/README.md (hunk 14 above)
Comment (n-clinton) [reply models/model_2025b/README.md:229]: Not sure which one you're talking about, but all those limitations are definitely present.

File: models/model_2025b/README.md (hunk 15 above)
Comment (n-clinton) [reply models/model_2025b/README.md]: Done.

File: models/model_2025b/README.md (hunk 3 above)
Comment (brycecr) [reply models/model_2025b/README.md]: Ah, got it

File: models/model_2025b/README.md (hunk 14 above)
Comment (brycecr) [reply models/model_2025b/README.md:229]: I was thinking about the artifacts mention, but what you say is fair. Let's leave it in

File: models/model_2025b/README.md (hunk 10 above)
Comment (brycecr) [reply models/model_2025b/README.md:194]: sorry my mistake. resolving

File: models/model_2025b/README.md (hunk 7 above)
Comment (brycecr) [reply models/model_2025b/README.md:113]: Oh, I see. There's a Descals reference for the dataset from the Sheil paper. This seems fine as is

**Review by brycecr — APPROVED** [review]
